packages/core/src/hydration/event_replay.ts

```diff
@@ -135,7 +132,9 @@ export function withEventReplay(): Provider[] {
               whenStable(appRef).then(() => {
                 const eventContractDetails = injector.get(JSACTION_EVENT_CONTRACT);
                 initEventReplay(eventContractDetails, injector);
-                removeListenersFromBlocks([''], injector);
+                const jsActionMap = injector.get(JSACTION_BLOCK_ELEMENT_MAP);
+                jsActionMap.get('')?.forEach(removeListeners);
```

Comment:
As an idea: I think it'd be great to introduce a const, which would help with code readability, for example:
```suggestion
                jsActionMap.get(EAGER_CONTENT_LISTENERS_KEY)?.forEach(removeListeners);
```